librz/cons/dietline.c

```diff
@@ -431,6 +474,186 @@ RZ_API int rz_line_hist_chop(const char *file, int limit) {
 	return 0;
 }
 
+RZ_API int rz_line_undo_free(void) {
+	rz_vector_free(I.undo_vec);
+	I.undo_vec = NULL;
+	I.undo_cursor = 0;
+	return true;
+}
+
+/* If possible, concatenate input characters according to the behavior of bash. (Others such as zsh don't do that) */
+static bool concat_undo_entry(const char *diff, const int diff_len) {
+	if (I.undo_vec->len && I.undo_cursor == I.undo_vec->len) {
```

Comment:
If you flip the condition, you could use early return pattern and avoid nesting.